tests/e2e/nvidia-container-toolkit_test.go

```diff
@@ -297,4 +298,51 @@ var _ = Describe("docker", Ordered, ContinueOnFailure, func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 	})
+
+	When("Running a container where the firmware folder resolves outside the container root", Ordered, func() {
+		var outputDir string
+		BeforeAll(func(ctx context.Context) {
+			output, _, err := runner.Run("mktemp -d -p $(pwd)")
+			Expect(err).ToNot(HaveOccurred())
+			outputDir = strings.TrimSpace(output)
+
+			_, _, err = runner.Run("docker pull ubuntu")
+			Expect(err).ToNot(HaveOccurred())
+
+			dockerBuildCmd := fmt.Sprintf(`docker build -t firmware-test --build-arg RM_VERSION="$(basename $(ls -d /lib/firmware/nvidia/*.*))" --build-arg CURRENT_DIR=%q - <<EOF`, outputDir)
+			dockerBuildDockerfile := `
+FROM ubuntu
+RUN mkdir -p /lib/firmware/nvidia/
+ARG RM_VERSION
+ARG CURRENT_DIR
+RUN ln -s /../../../../../../../../\$CURRENT_DIR /lib/firmware/nvidia/\$RM_VERSION
+EOF`
+			_, _, err = runner.Run(dockerBuildCmd + dockerBuildDockerfile)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterAll(func(ctx context.Context) {
+			if outputDir != "" {
+				runner.Run(fmt.Sprintf("rm -rf %s", outputDir))
+			}
+		})
+
+		It("should not fail when using CDI", func(ctx context.Context) {
+			output, _, err := runner.Run("docker run --rm --runtime=nvidia --gpus=all firmware-test")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output).To(BeEmpty())
+
+			output, _, _ = runner.Run(fmt.Sprintf("ls -A %s", outputDir))
+			Expect(output).To(BeEmpty())
+		})
+
+		It("should fail when using the nvidia-container-runtime-hook", Label("legacy"), func(ctx context.Context) {
+			_, stderr, err := runner.Run("docker run --rm --runtime=runc --gpus=all firmware-test")
+			Expect(err).To(HaveOccurred())
+			Expect(stderr).To(ContainSubstring("nvidia-container-cli.real: mount error: path error:"))
```

Comment:
Let's add a test for the `nvidia-container-runtime` too:

```suggestion
        It("should not fail when using the nvidia-container-runtime-hook", func(ctx context.Context) {
            _, _, err := runner.Run("docker run --rm --runtime=nvidia -e NVIDIA_VISIBLE_DEVICES=all -e NVIDIA_DRIVER_CAPABILITIES=all firmware-test")
        }

		It("should fail when using the nvidia-container-runtime-hook", Label("legacy"), func(ctx context.Context) {
```